importers/rules.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importers
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/url"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	plabels "github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/rules"
+	"github.com/prometheus/prometheus/tsdb/importer/blocks"
+)
+
+// RuleImporter is the importer for rules
+type RuleImporter struct {
+	logger log.Logger
+
+	config RuleConfig
+	groups map[string]*rules.Group
+
+	apiClient v1.API
+
+	writer *blocks.MultiWriter
+}
+
+// RuleConfig is the config for the rule importer
+type RuleConfig struct {
+	Start        string
+	End          string
+	EvalInterval time.Duration
+	URL          string
+}
+
+// NewRuleImporter creates a new rule importer
+func NewRuleImporter(logger log.Logger, config RuleConfig) *RuleImporter {
+	return &RuleImporter{
+		config: config,
+	}
+}
+
+// Init initializes the rule importer which creates a new block writer
+// and creates an Prometheus API client
+func (importer *RuleImporter) Init() error {
+	// create new block writer
+	newBlockDir, err := ioutil.TempDir("", "rule_blocks")
```

Comment:
/tmp may not be big enough, I'd suggest working in the output directory.

When do you clean this up?

Reply:
what is "the output dir"? should I take in a path as an argument where the blocks will be written?

Reply:
Yes, wherever you're writing the output.